zebra/rt_netlink.c

```diff
@@ -2864,7 +2873,8 @@ ssize_t netlink_nexthop_msg_encode(uint16_t cmd,
 					nl_attr_nest_end(&req->n, nest);
 				}
 
-				if (!sid_zero(&nh->nh_srv6->seg6_segs)) {
+				if (nh->nh_srv6->seg6_segs &&
+				    !sid_zero(nh->nh_srv6->seg6_segs)) {
```

Comment:
And this is exactly why returning true is confusing when it's a null pointer, we have to do these shenanigans to get the right value.

Reply:
I addressed a function "sid_zero()". the "TRUE" return when pointer is NULL is replaced with assert. The logic to verify if pointer outside of the "sid_zero()". So it should be compliant with what you expect.